api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -94,6 +92,15 @@ type KonnectExtensionSpec struct {
 	DataPlaneLabels []DataPlaneLabel `json:"dataPlaneLabels,omitempty"`
 }
 
+// KonnectExtensionControlPlane is the configuration for the Konnect Control Plane.
+type KonnectExtensionControlPlane struct {
```

Comment:
Can you provide some examples of fields that would land in `KonnectExtensionControlPlane` which wouldn't fit well in the `KonnectExtensionSpec`?

Reply:
An immediate example is the type of provisioning:
- context: running in KO (no more KIC)
- create a `GatewayConfiguration` and attach a `KonnectExtension`
- set `provisioning: automatic` in the `KonnectExtension` `ControlPlane` section
- the `Gateway`s using that GatewayConfiguration will result in a pair of `KonnectGatewayControlPlane` and a `DataPlane`. The `KonnectGatewayControlPlane` is provisioned automatically, instead of being provided by the user. All the information to do it (`KonnectAPIAuthConfiguration` reference) is already in the `KonnectExtension`

Does it make sense to you?

Reply:
Since the type of `ControlPlane` provisioning is a `ControlPlane`-related parameter, having it in a dedicated `controlPlane`-related part of the spec, along with the `ControlPlaneRef` would make more sense than having it in the outer spec IMO.

Reply:
That makes sense, thanks 👍